### DISCUSSION TITLE
Arrow Dataset as an extension API?

### DISCUSSION THREAD

**Discussion by meghprkh** [duckdb/duckdb#14642] [Q&A]
Would it be possible to use the arrow dataset API as an extension building API connecting to a (scan-only?) data source?

It does seem to support filters and sorts, and would be a very portable way to implement a data source! Though I can not find documentation on how to reject or accept a filter pushdown?

I also understand that this is supported for Python already (not sure about the extent of pushdowns) and was wondering if this could be supported in C++ too?

Maybe we register a "dataset factory" as say `scan_mydb` and allow it to specify a list of arguments that are passed onwards to a specific dataset creator?

**Comment by Tishj** [comment]
In python we communicate arrow through the C data interface, you can make use of these same methods in the C++ API, find the "arrow_scan" method and craft some pointer parameters to pass to it
